src/node/internal/internal_tls_wrap.ts

```diff
@@ -0,0 +1,644 @@
+  // record fails, the handshake done info callbacks does not occur, and the
+  // socket closes.
+  // * 1.3: The OCSP response comes in the same record that finishes handshake,
+  // so even after .destroy(), the handshake done info callback occurs
+  // immediately after onocspresponse(). Ignore it.
+  if (!this._handle) return;
+
+  this._secureEstablished = true;
+  if (
+    this._tlsOptions.handshakeTimeout &&
+    this._tlsOptions.handshakeTimeout > 0
+  )
+    this.setTimeout(0, this._handleTimeout.bind(this));
+  this.emit('secure');
+};
+
+TLSSocket.prototype._start = function _start(this: TLSSocket): void {
+  if (this.connecting) {
+    this.once('connect', this._start.bind(this));
+    return;
+  }
+
+  // Socket was destroyed before the connection was established
+  if (!this._handle) return;
+
+  // this._handle.start();
+};
+
+TLSSocket.prototype.setServername = function setServername(
+  this: TLSSocket,
+  name: string
+): void {
+  validateString(name, 'name');
+  // Does not apply to Cloudflare Workers.
+};
+
+TLSSocket.prototype.setSession = function (_session: string | Buffer): void {
+  // Do nothing. We don't support setting session.
+};
+
+// @ts-expect-error TS2322 Inconsistencies between @types/node
+TLSSocket.prototype.getPeerCertificate = function (
+  _detailed?: boolean
+): ReturnType<TLSSocketType['getPeerCertificate']> {
+  // Returns an object representing a peer certificate.
+  // This function is not supported.
+  throw new Error('getPeerCertificate is not implemented');
+};
+
+TLSSocket.prototype.getCertificate = function (
+  this: TLSSocket
+): ReturnType<TLSSocketType['getCertificate']> {
+  // Returns an object representing the local certificate.
+  // This function is not supported.
+  return null;
+};
+
+TLSSocket.prototype.getPeerX509Certificate = function (
+  this: TLSSocket,
+  _detailed?: boolean
+): ReturnType<TLSSocketType['getPeerX509Certificate']> {
+  // Returns the peer certificate as an X509 certificate.
+  // This function is not supported.
+  return undefined;
+};
+
+TLSSocket.prototype.getX509Certificate = function (
+  this: TLSSocket
+): ReturnType<TLSSocketType['getX509Certificate']> {
+  // Returns the local certificate as an X509 certificate.
+  // This function is not supported.
+  return undefined;
+};
+
+TLSSocket.prototype.setKeyCert = function (
+  this: TLSSocket,
+  _context: unknown
+): void {
+  // Changing private key and certificate to be used with TCP connection
+  // is not supported due to the limitations of connect() api.
+};
+
+[
+  'getCipher',
+  'getSharedSigalgs',
+  'getEphemeralKeyInfo',
+  'getFinished',
+  'getPeerFinished',
+  'getProtocol',
+  'getSession',
+  'getTLSTicket',
+  'isSessionReused',
+  'enableTrace',
+].forEach((method) => {
+  // @ts-expect-error TS7053 Omitting...
+  TLSSocket.prototype[method] = function (): null {
+    // None of these functions are supported by connect() api.
+    return null;
```

Comment:
It's not clear why these are handled like this but the `getX509Certificate`, etc cases above are separated out.

Reply:
This is directly from Node.js. It is separated originally in the codebase because they are all properties of this._handle[xxx]

Reply:
A comment to that effect would be helpful